addons/account_peppol/models/account_move.py

```diff
@@ -29,8 +30,20 @@ def action_cancel_peppol_documents(self):
         if any(move.peppol_move_state in {'processing', 'done'} for move in self):
             raise UserError(_("Cannot cancel an entry that has already been sent to PEPPOL"))
         self.peppol_move_state = 'canceled'
+        self.send_and_print_values = False
 
     @api.depends('peppol_message_uuid')
     def _compute_peppol_is_demo_uuid(self):
         for move in self:
             move.peppol_is_demo_uuid = (move.peppol_message_uuid or '').startswith('demo_')
+
+    @api.depends('partner_id', 'state', 'partner_id.account_peppol_is_endpoint_valid', 'company_id.account_peppol_proxy_state')
```

Comment:
That depends are bad. You don't want to force the orm to write on thousands invoices when `account_peppol_is_endpoint_valid` changed nor `account_peppol_proxy_state`

Reply:
Btw, depends on `partner_id` is useless if you already have a dependency to `partner_id.some_field`